core/src/test/scala/integration/kafka/api/TransactionsTest.scala

```diff
@@ -112,6 +112,53 @@ class TransactionsTest extends KafkaServerTestHarness {
   }
 
   @Test
+  def testReadCommittedConsumerShouldNotSeeUndecidedData(): Unit = {
+    val producer1 = transactionalProducers.head
+    val producer2 = TestUtils.createTransactionalProducer("other", servers)
```

Comment:
We are leaking this producer. Is there a reason why you cannot use one of the existing `transactionalProducers`?